### DISCUSSION TITLE
Set Screen/Display resolution?

### DISCUSSION THREAD

**Discussion by mundorfd** [ttscoff/bunch-docs#209] [Q&A]
I want to create a Zoom bunch that scales my main display resolution. My large screen is hard to see when I am sharing, so during meetings I would like to scale it to 1920x800 for instance, instead of the default for my display which is 3840x1600. I looked for a Keyboard Maestro action for this as well, and didn't find any. Maybe an apple script? Any help appreciated!

**Comment by ttscoff** [comment]
Hm, I don't know offhand of any way to automate screen resolution changes. If any utilities exist to do this, it's a near certainty that they could be incorporated with Bunch, but you'd have to do some googling to see if one exists.

**Reply by mundorfd** [reply]
I found one called "Display Menu." Their Pro version, for $1.99, has AppleScript support. As a result - this does the trick: 

```AppleScript
tell application "Display Menu"
	set theResolution to current resolution on display "DELL U3818DW"
	# set theDialogText to theResolution
	# display dialog theDialogText
	if (theResolution = "3840 x 1600 30.0 Hz non-Retina") then
		select resolution "1920 x 800 30.0 Hz Retina" on display "DELL U3818DW"
	else
		select resolution "3840 x 1600 30.0 Hz non-Retina" on display "DELL U3818DW"
	end if
	
end tell
```
The commented lines helped me get the right syntax for the resolutions.

As I recall, I can run AppleScript from bunch.

**Comment by ttscoff** [comment]
Yep, you can save that as a script and run it with a * line, or embed the whole thing using heredoc syntax.

https://bunchapp.co/docs/bunch-files/scripts/applescript/#heredoc

-Brett
________________________________
